### PR TITLE
EASY-2055 generate bag-store.bag-id at submit

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes EASY-2055 

#### When applied it will
* generate a new `bag-store.bag-id` at each (re)submit
* use this ID as directory in ingest-flow-inbox
* [x] don't add a second value to the desposit.properties, but overwrite a previous

#### Related pull requests on github
repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-                      | [PR#](PRlink)     | ..

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.deposit/Errors.scala [hunk 1]
```diff
@@ -83,16 +83,11 @@ object Errors extends DebugEnhancedLogging {
...
-   * The end user can compare the UUID with the URL of a deposit.
-   * The UUID can help communication with trouble shooters.
-   */
+  /** Note: submit area == easy-ingest-flow-inbox */
   case class AlreadySubmittedException(uuid: UUID)
     extends ServletResponseException(
-      CONFLICT_409,
-      s"The deposit (UUID $uuid) already exists in the submit area. Possibly due to a resubmit."
+      INTERNAL_SERVER_ERROR_500,
+      s"The submit-id (UUID $uuid) already exists in the submit area."
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.deposit/Errors.scala]: This text is possibly shown to a user. Give a less technical error message, please.

File: src/main/scala/nl.knaw.dans.easy.deposit/Submitter.scala [hunk 2]
```diff
@@ -64,9 +64,9 @@ class Submitter(stagingBaseDir: File,
    * and moving that copy to the submit-to area.
    *
    * @param draftDeposit the deposit object to submit
-   * @return
+   * @return the UUID of the deposit in the ingest-flow-inbox
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.deposit/Submitter.scala]:
```suggestion
   * @return the UUID of the deposit in the submit area (easy-ingest-flow-inbox)
```

File: src/test/scala/nl.knaw.dans.easy.deposit/servlets/IntegrationSpec.scala [hunk 3]
```diff
@@ -354,8 +354,12 @@ class IntegrationSpec extends TestSupportFixture with ServletFixture with Scalat
       body shouldBe ""
       status shouldBe NO_CONTENT_204
 
+      val bagStoreBagId = new PropertiesConfiguration((depositDir / "deposit.properties").toJava)
+        .getString("bag-store.bag-id")
+      bagStoreBagId shouldNot be(null)
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/servlets/IntegrationSpec.scala]:
```suggestion
      bagStoreBagId should not be null
```

Comment (rvanheest) [reply src/test/scala/nl.knaw.dans.easy.deposit/servlets/IntegrationSpec.scala]: or even better, use `new PropertiesConfiguration((depositDir / "deposit.properties").toJava).containsKey("bag-store.bag-id") shouldBe true`

Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/servlets/IntegrationSpec.scala]: if you test this property, it probably means that you expect `bagStoreBagId` not to be present earlier. Could you add that test as well, please?

File: src/test/scala/nl.knaw.dans.easy.deposit/SubmitterSpec.scala [hunk 4]
```diff
@@ -103,12 +104,38 @@ class SubmitterSpec extends TestSupportFixture with MockFactory {
...
+    val bagStoreBagId2 = succeedingSubmit(depositDir)
+
+    // we only have the second bag-store.bag-id
+    bagStoreBagId1 shouldNot be(bagStoreBagId2)
+    val lines = depositProps.contentAsString.split("\n")
+    lines.count(_.contains("bag-store.bag-id")) shouldBe 1
+    lines.count(_ == s"bag-store.bag-id = $bagStoreBagId2") shouldBe 1
+  }
+
+  private def succeedingSubmit(deposit: DepositDir) = {
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/SubmitterSpec.scala]:
```suggestion
  private def succeedingSubmit(deposit: DepositDir): String = {
```

File: src/test/scala/nl.knaw.dans.easy.deposit/SubmitterSpec.scala [hunk 5]
```diff
@@ -103,12 +104,38 @@ class SubmitterSpec extends TestSupportFixture with MockFactory {
     val depositDir = createDeposit(datasetMetadata.copy(messageForDataManager = None))
     addDoiToDepositProperties(getBag(depositDir))
 
-    createSubmitter(userGroup).submit(depositDir) shouldBe a[Success[_]]
+    val bagStoreBagId1 = succeedingSubmit(depositDir)
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/SubmitterSpec.scala]: why this `1` in `bagStoreBagId1`?

File: src/test/scala/nl.knaw.dans.easy.deposit/SubmitterSpec.scala [hunk 6]
```diff
@@ -103,12 +104,38 @@ class SubmitterSpec extends TestSupportFixture with MockFactory {
...
 
+  it should "overwrite a previous bag-store.bag-id at resubmit" in {
+
+    val depositDir = createDeposit(datasetMetadata.copy(messageForDataManager = None))
+    addDoiToDepositProperties(getBag(depositDir))
+
+    // submit twice
+    val bagStoreBagId1 = succeedingSubmit(depositDir)
+    val depositProps = depositDir.bagDir.parent / "deposit.properties"
+    depositProps.write(depositProps.contentAsString.replace("SUBMITTED", "DRAFT"))
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/SubmitterSpec.scala]: shouldn't `"DRAFT"` be `"REJECTED"`? Don't remember what the picture by @lindareijnhoudt said.

File: src/test/scala/nl.knaw.dans.easy.deposit/SubmitterSpec.scala [hunk 7]
```diff
@@ -103,12 +104,38 @@ class SubmitterSpec extends TestSupportFixture with MockFactory {
...
+    val bagStoreBagId2 = succeedingSubmit(depositDir)
+
+    // we only have the second bag-store.bag-id
+    bagStoreBagId1 shouldNot be(bagStoreBagId2)
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/SubmitterSpec.scala]:
```suggestion
    bagStoreBagId1 should not be bagStoreBagId2
```

File: src/test/scala/nl.knaw.dans.easy.deposit/servlets/IntegrationSpec.scala [hunk 8]
```diff
@@ -206,7 +207,7 @@ class IntegrationSpec extends TestSupportFixture with ServletFixture with Scalat
   "scenario: create - ... - sumbit" should "create submitted dataset copied from a draft" in {
     val uuid: String = setupSubmittedDeposit
 
-    // resubmit fails
+    // resubmit succeeds
     val props = testDir / s"drafts/foo/$uuid/deposit.properties"
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/servlets/IntegrationSpec.scala]:
```suggestion
     val props = testDir / "drafts" / "foo" / uuid / "deposit.properties"
```

File: src/main/scala/nl.knaw.dans.easy.deposit/Errors.scala (hunk 1 above)
Comment (jo-pol) [reply src/main/scala/nl.knaw.dans.easy.deposit/Errors.scala]: No, it won't:
https://github.com/DANS-KNAW/easy-deposit-api/blob/f004b027d0766a7cac77dac6ca8f43e021128715/src/main/scala/nl.knaw.dans.easy.deposit/Errors.scala#L47-L54

File: src/test/scala/nl.knaw.dans.easy.deposit/SubmitterSpec.scala (hunk 6 above)
Comment (lindareijnhoudt) [reply src/test/scala/nl.knaw.dans.easy.deposit/SubmitterSpec.scala]: i agree. 
image
REJECTED -> DRAFT is when the user opens it and saves it

File: src/test/scala/nl.knaw.dans.easy.deposit/servlets/IntegrationSpec.scala (hunk 3 above)
Comment (jo-pol) [reply src/test/scala/nl.knaw.dans.easy.deposit/servlets/IntegrationSpec.scala]: Need `bagStoreBagId` on the next line

Comment (rvanheest) [reply src/test/scala/nl.knaw.dans.easy.deposit/servlets/IntegrationSpec.scala]: then keep the value, but use `.containsKey`

**Pull request merged**